### ISSUE TITLE
Update to be compatible with Node.js 0.12

### ISSUE THREAD

**Issue by tryIO** [DefinitelyTyped/tsd#144]
Hi there! I just updated my Node to version 0.12, but it seems tsd is not compatible with it, any plan for this (maybe in the next version 6.5, but I wasn't able to find anything about it)? or any workaround?

**Comment by toedter** [comment]
I would also be interested in running tsd with node 0.12.x

**Comment by blakeembrey** [comment]
@tryIO What version are you currently using? I haven't had any troubles, but I've mostly been running `0.6`.

**Comment by tryIO** [comment]
I used 0.12.0 and when I tried to install "tsd" it replied me that it needs NodeJS version greater than 0.6 and lesser than 0.11. I didn't re-tried recently, but I will do to check if everything is still the same.

**Comment by Linicks** [comment]
Here is a list of the warnings that I had during a recent install:

```
npm install tsd -g
npm WARN engine joi-assert@0.0.3: wanted: {"node":">= 0.10.0 <= 0.11.0"} (current: {"node":"2.0.0","npm":"2.9.0"})
npm WARN peerDependencies The peer dependency minichain@~0.0.1 included from minitable will no
npm WARN peerDependencies longer be automatically installed to fulfill the peerDependency 
npm WARN peerDependencies in npm 3+. Your application will need to depend on it explicitly.
```

**Comment by blakeembrey** [comment]
@Linicks Installing `tsd` should now be installing `0.6.0` instead of `0.5.x` and it should fix this issue.

**Comment by QuantumInformation** [comment]
still doesn't work

**Comment by blakeembrey** [comment]
@QuantumInformation You're right, looks like someone already logged an issue on that package (https://github.com/Bartvds/joi-assert/issues/2). However, I'm not sure if it'll be updated so the best action might be to refactor and bring it inline with `0.6.x`.

**Comment by QuantumInformation** [comment]
How do I do that? @blakeembrey

**Comment by blakeembrey** [comment]
@QuantumInformation You're welcome to submit a pull request, it'll mostly just be copying and pasting the code into this codebase and replacing previous references of `joi-assert` to the new code. TSD will still work with this warning, since the code in that module doesn't actually have a dependency on those node versions.

**Comment by QuantumInformation** [comment]
tsd fails to install for me:
`npm install tsd -g`

If I run tsd in terminal I get:

tsd command not found

**Comment by blakeembrey** [comment]
@QuantumInformation It wouldn't be caused by this. Did you get any errors during installation? Do other node modules work for you (maybe you're bash path is incorrect)?

**Comment by QuantumInformation** [comment]
other node modules work fine, I didn't get errors, this is what happened on install:

![image](https://cloud.githubusercontent.com/assets/216566/8251105/a05595e2-1671-11e5-8ce5-058c0021a242.png)

**Comment by blakeembrey** [comment]
@QuantumInformation So your node modules are being installed globally to `/Users/nikos/.node/bin` and you need to make sure that's in your path (`echo $PATH`). Have you tried installing any other modules globally and using their CLIs? From that output, you can see installation is successful - it's just the command is missing and I would say it's your path. You'll need edit your `.bash_profile` and add `export PATH=/Users/nikos/.node/bin:$PATH` to the end for it to work.

**Comment by QuantumInformation** [comment]
my path:

`/Users/nikos/.npm-packages/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin`

`bash_profile` :
```NPM_PACKAGES="/Users/nikos/.npm-packages"
NODE_PATH="$NPM_PACKAGES/lib/node_modules:$NODE_PATH"
PATH="$NPM_PACKAGES/bin:$PATH"
```

**Comment by QuantumInformation** [comment]
adding 
export PATH=/Users/nikos/.node/bin:$PATH

got tsd working, I reckon my other cli modules got installed somewhere else.
thanks!

**Comment by blakeembrey** [comment]
@QuantumInformation No worries :smile: Looks like your package directory has changed from `.node` to `.npm-packages` at some point. Maybe check that out since your packages might start being installed differently to what you're used to. I don't think I can help you too much there, but start by checking `npm config --global get prefix` and `echo $NPM_CONFIG_PREFIX` - perhaps its been changed somewhere?

**Comment by QuantumInformation** [comment]
Thanks, this is what I have: 
![image](https://cloud.githubusercontent.com/assets/216566/8253257/01c14592-1686-11e5-94f6-05c237649a02.png)

**Comment by drozzy** [comment]
The same is happening to me. `tsd: command not found` after the ```npm WARN engine joi-assert@0.0.3: wanted: {"node":">= 0.10.0 <= 0.11.0"} (curren
t: {"node":"0.12.6","npm":"2.11.2"})```
